src/main/kotlin/com/demonwav/mcdev/platform/mixin/config/reference/CompatibilityLevel.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.mixin.config.reference
+
+import com.demonwav.mcdev.platform.mixin.util.MixinConstants.Classes.COMPATIBILITY_LEVEL
+import com.demonwav.mcdev.util.reference.InspectionReference
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiEnumConstant
+import com.intellij.psi.PsiReference
+import com.intellij.psi.PsiReferenceBase
+import com.intellij.psi.PsiReferenceProvider
+import com.intellij.util.ArrayUtil
+import com.intellij.util.ProcessingContext
+
+object CompatibilityLevel : PsiReferenceProvider() {
+
+    override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<PsiReference> = arrayOf(Reference(element))
+
+    private class Reference(element: PsiElement) : PsiReferenceBase<PsiElement>(element), InspectionReference {
+
+        override val description
+            get() = "compatibility level '%s'"
+
+        override val unresolved
+            get() = resolve() == null
+
+        override fun resolve(): PsiElement? {
+            val compatibilityLevel = findCompatibilityLevel(element)
+            return compatibilityLevel?.findFieldByName(value, false)?.takeIf { it is PsiEnumConstant }
```

Comment:
Should this be cached? Seems like the state of this wouldn't change once it's created. Could be easily done using a lazy delegate.